### ISSUE TITLE
AWS::CloudFront::KeyGroup

### ISSUE THREAD

**Issue by ctd** [aws-cloudformation/cloudformation-coverage-roadmap#687]
## Scope of request 

tldr: New resource type for an existing service is desired

Amazon CloudFront [launched the key groups feature](https://aws.amazon.com/about-aws/whats-new/2020/10/cloudfront-iam-signed-url/) on Oct 22, 2020. Key groups can currently be created and managed via the API or Console.

This request is to create a KeyGroup resource type to use this feature via CloudFormation.

#686 exists for adding support to use key groups (once created) in a Distribution.

## Expected behavior

- In Create, the resource should affect a corresponding [CreateKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_CreateKeyGroup.html) API call.
- In Update: [UpdateKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_UpdateKeyGroup.html)
- In Delete: [DeleteKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_DeleteKeyGroup.html)

The resource type should have a return value, or attribute, suitable to be used in a Distribution CacheBehavior (see #686).

## 5. Helpful Links to speed up research and evaluation

- [Feature announcement](https://aws.amazon.com/about-aws/whats-new/2020/10/cloudfront-iam-signed-url/)
- [Feature documentation](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-trusted-signers.html#choosing-key-groups-or-AWS-accounts)
- API documentation for:
  - [CreateKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_CreateKeyGroup.html)
  - [UpdateKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_UpdateKeyGroup.html)
  - [DeleteKeyGroup](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_DeleteKeyGroup.html)
- [An equivalent request for the Terraform provider](https://github.com/hashicorp/terraform-provider-aws/issues/15912)

## Category

Networking & Content (CloudFront)

**Comment by bbalka** [comment]
@ctd We supporting following operations using CFN

In Create, the resource should affect a corresponding CreateKeyGroup API call.
In Update: UpdateKeyGroup
In Delete: DeleteKeyGroup
